### PR TITLE
SONARJAVA-1234 handle JPA EntityManager

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/SQLInjectionCheck.java [hunk 1]
```diff
@@ -42,11 +42,30 @@
 @SqaleConstantRemediation("20min")
 public class SQLInjectionCheck extends AbstractInjectionChecker {
 
+  private static final MethodMatcher HIBERNATE_SESSION_CREATE_QUERY_MATCHER = MethodMatcher.create()
+    .callSite(TypeCriteria.subtypeOf("org.hibernate.Session"))
+    .name("createQuery") // method from the interface org.hibernate.SharedSessionContract, implemented by org.hibernate.Session
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/SQLInjectionCheck.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by benzonico** [comment]
Not related to this work but shouldn't this rule benefit from precise issue location as well ?

File: java-checks/src/main/java/org/sonar/java/checks/SQLInjectionCheck.java [hunk 2]
```diff
@@ -42,11 +42,31 @@
 @SqaleConstantRemediation("20min")
 public class SQLInjectionCheck extends AbstractInjectionChecker {
 
+  private static final MethodMatcher HIBERNATE_SESSION_CREATE_QUERY_MATCHER = MethodMatcher.create()
+    // method from the interface org.hibernate.SharedSessionContract, implemented by org.hibernate.Session
+    .callSite(TypeCriteria.subtypeOf("org.hibernate.Session"))
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/SQLInjectionCheck.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request merged**